### PR TITLE
Implementing pickle methods

### LINKED ISSUE #419: Pickling a Betterproto message
Hi,

I have found an issue when I wanted to use this library in combination with multiprocessing/pickling.
I am using version 2.0.0b5

When the resulting message is send back by one of my running processes with the `map` function it changes empty values and also when I only run pickle they are changed.

The message:
```
syntax = "proto3";

message TestMessage {
    string value = 1;
}
```

To code:
```
test_message = TestMessage()
print(test_message)
pickled = pickle.dumps(test_message)
unpickled = pickle.loads(pickled)
print(pickle.loads(pickled))
print(unpickled.value)
print(type(unpickled.value))
```

The output:
```
TestMessage()
TestMessage(value=<object object at 0x7febf0791fe0>)
<object object at 0x7fd9ef242ff0>
<class 'object'>
```

So it seems it adds an object instead of an empty value.

### PR DESCRIPTION
**Pull request by kevinaud**
Fixes #419

### CODE REVIEW

File: src/betterproto/__init__.py [hunk 1]
```diff
@@ -858,6 +858,12 @@ def SerializeToString(self: T) -> bytes:
         """
         return bytes(self)
 
+    def __getstate__(self):
+        return bytes(self)
+
+    def __setstate__(self, pickled_bytes):
+        return self.parse(pickled_bytes)
```
Comment (Gobot1234) [inline src/betterproto/__init__.py:865]:
```suggestion
    def __getstate__(self) -> bytes:
        return bytes(self)

    def __setstate__(self: T, pickled_bytes: bytes) -> T:
        return self.parse(pickled_bytes)
```

**Comment by Alexejhero** [comment]
Any update on this? We're running in the same problem

**Comment by petrastegaev** [comment]
Hi! Please merge, this bug is blocking me.

**Comment by Gobot1234** [comment]
I'm not planning on merging this until tests pass and the current failures which I'm not quite sure about failing because copy doesn't seem to use these pickle methods for shallow copies.

**Comment by kilimnik** [comment]
I had an issue using the unpickled objects, what fixed it for me, was calling the `__post_init__` inside the `__setstate__` function after parsing.

**Comment by Gobot1234** [comment]
@kevinaud if you want this merging you need to make this branch editable to contributors or merge the PR I've made to your branch

**Comment by cetanu** [comment]
Work continuing in #535

**Pull request closed without merging**